Lib/collections/__init__.py

```diff
@@ -240,11 +240,18 @@ def pop(self, key, default=__marker):
         is raised.
 
         '''
-        if key in self:
-            result = self[key]
-            del self[key]
+        marker = self.__marker
+        result = dict.pop(self, key, marker)
+        if result is not marker:
+            link = self.__map.pop(key)
+            link_prev = link.prev
+            link_next = link.next
+            link_prev.next = link_next
+            link_next.prev = link_prev
+            link.prev = None
+            link.next = None
             return result
-        if default is self.__marker:
+        if default is marker:
```

Comment:
Is this present in CPython?

Reply:
Yep https://github.com/python/cpython/blob/main/Lib/collections/__init__.py#L246

Reply:
ah, so this bug surfaced because we update the test lib but not the `collections`? Pretty nice if so.